### ISSUE TITLE
Creating account error. Unknown column 'temperature_scale'

### ISSUE THREAD

**Issue by brusapa** [monicahq/monica#2243]
**Describe the bug**
I have deployed monica through docker-compose, when I try to sign up my first user I get this error:
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'temperature_scale' in 'field list' (SQL: insert into `users` (`account_id`, `first_name`, `last_name`, `email`, `password`, `created_at`, `locale`, `timezone`, `currency_id`, `temperature_scale`, `updated_at`) values (2, MyName, MySurname, mymail@mymailprovider.com, MyPasswordHash, 2019-01-02 08:58:31, en, America/Chicago, 2, fahrenheit, 2019-01-02 08:58:31))
```

**Docker-Compose file**
```
version: '3'
services:
  monicahq:
    image: monicahq/monicahq
    links:
      - mysql
    expose:
      - 80
    env_file: .env
    volumes:
      - /MyLocalStoragePath/monica/storage:/var/www/monica/storage
    restart: always
    labels:
      - "traefik.docker.network=proxynet"
      - "traefik.enable=true"
      - "traefik.monica.frontend.rule=Host:monica.MyDomain.com"
      - "traefik.monica.port=80"
      - "traefik.monica.protocol=http"
    networks:
      - postfixnet
      - proxynet
      - monicanet

  mysql:
    image: mysql:5.7
    container_name: monicahq-mysql
    environment:
      - MYSQL_ROOT_PASSWORD=${DB_ROOT_PASSWORD}
      - MYSQL_DATABASE=${DB_DATABASE}
      - MYSQL_USER=${DB_USERNAME}
      - MYSQL_PASSWORD=${DB_PASSWORD}
    volumes:
      - /MyLocalStoragePath/monica/sql:/var/lib/mysql
    restart: always
    networks:
      - monicanet

networks:
  postfixnet:
    external: true
  proxynet:
    external: true
  monicanet:
    driver: bridge
```

** .env file **
```
# Two choices: local|production. Use local if you want to install Monica as a
# development version. Use production otherwise.
APP_ENV=production

# true if you want to show debug information on errors. For production, put this
# to false.
APP_DEBUG=false

# The encryption key. This is the most important part of the application. Keep
# this secure otherwise, everyone will be able to access your application.
# Must be 32 characters long exactly.
# Use `php artisan key:generate` to generate a random key.
APP_KEY=MyKey1

# Prevent information leakage by referring to IDs with hashIds instead of
# the actual IDs used in the database.
HASH_SALT=ChangeMeBy20+KeyLength
HASH_LENGTH=18

# The URL of your application.
APP_URL=https://monica.MyDomain.com

# Database information
# To keep this information secure, we urge you to change the default password
DB_CONNECTION=mysql
DB_HOST=monicahq-mysql
DB_PORT=3306
# You can use mysql unix socket if available, it overrides DB_HOST and DB_PORT values.
#DB_UNIX_SOCKET=/var/run/mysqld/mysqld.sock
DB_ROOT_PASSWORD=MyKey2
DB_DATABASE=monica
DB_USERNAME=monica
DB_PASSWORD=MyKey3
DB_PREFIX=
DB_TEST_HOST=127.0.0.1
DB_TEST_DATABASE=monica_test
DB_TEST_USERNAME=homestead
DB_TEST_PASSWORD=MyKey4

# Use utf8mb4 database charset format to support emoji characters
# ⚠ be sure your DBMS supports utf8mb4 format
DB_USE_UTF8MB4=true

# Mail credentials used to send emails from the application.
MAIL_DRIVER=smtp
MAIL_HOST=postfix
MAIL_PORT=25
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS=monicahq@MyDomain.com
MAIL_FROM_NAME="Monica HQ"
APP_EMAIL_NEW_USERS_NOTIFICATION=monicahq@MyDomain.com

# Default locale used in the application.
APP_DEFAULT_LOCALE=es

# Ability to disable signups on your instance.
# Can be true or false. Default to false.
APP_DISABLE_SIGNUP=false

# Enable user email verification.
APP_SIGNUP_DOUBLE_OPTIN=false

# Set trusted proxy IP addresses.
# To trust all proxies that connect directly to your server, use a "*".
# To trust one or more specific proxies that connect directly to your server, use a comma separated list of IP addresses.
APP_TRUSTED_PROXIES=*

# Enable automatic cloudflare trusted proxy discover
APP_TRUSTED_CLOUDFLARE=false

# Frequency of creation of new log files. Logs are written when an error occurs.
# Refer to config/logging.php for the possible values.
LOG_CHANNEL=daily

# Error tracking. Specific to hosted version on .com. You probably don't need
# those.
SENTRY_SUPPORT=false
SENTRY_LARAVEL_DSN=

# Send a daily ping to https://version.monicahq.com to check if a new version
# is available. When a new version is detected, you will have a message in the
# UI, as well as the release notes for the new changes. Can be true or false.
# Default to true.
CHECK_VERSION=true

# Change this only if you know what you are doing
CACHE_DRIVER=database
SESSION_DRIVER=file
SESSION_LIFETIME=120
QUEUE_DRIVER=sync

# Maximum allowed size for uploaded files, in kilobytes.
# Make sure this is an integer, without commas or spaces.
DEFAULT_MAX_UPLOAD_SIZE=10240

# Maximum allowed storage size per account, in megabytes.
# Make sure this is an integer, without commas or spaces.
DEFAULT_MAX_STORAGE_SIZE=512

# Default filesystem to store uploaded files.
# Possible values: public|s3
DEFAULT_FILESYSTEM=public

# AWS keys for S3 when using this storage method
AWS_KEY=
AWS_SECRET=
AWS_REGION=us-east-1
AWS_BUCKET=
AWS_SERVER=

# Allow Two Factor Authentication feature on your instance
MFA_ENABLED=false

# Enable CardDAV support (beta feature)
CARDDAV_ENABLED=true

# CLIENT ID and SECRET used for the official mobile application
# This is to make sure that only the mobile application that you approve can
# access the route to let your users sign in with their credentials
MOBILE_CLIENT_ID=
MOBILE_CLIENT_SECRET=

# Allow to access general statistics about your instance through a public API
# call
ALLOW_STATISTICS_THROUGH_PUBLIC_API_ACCESS=false

# Indicates that each user in the instance must comply to international policies
# like CASL or GDPR
POLICY_COMPLIANT=true

# Have access to paid features available on https://monicahq.com, for free.
# Can be true or false. Default to false.
# If set to true, that means your users will have to pay to access the paid
# features. We use Stripe to do this.
REQUIRES_SUBSCRIPTION=false

# Number of contacts allowed in the Free version if the instance requires
# a subscription
NUMBER_OF_ALLOWED_CONTACTS_FREE_ACCOUNT=10

# ONLY NECESSARY IF MONICA REQUIRES A SUBSCRIPTION TO WORK
# Leave blank unless you know what you are doing.
STRIPE_KEY=
STRIPE_SECRET=
PAID_PLAN_MONTHLY_FRIENDLY_NAME=
PAID_PLAN_MONTHLY_ID=
PAID_PLAN_MONTHLY_PRICE=
PAID_PLAN_ANNUAL_FRIENDLY_NAME=
PAID_PLAN_ANNUAL_ID=
PAID_PLAN_ANNUAL_PRICE=

# Enable geolocation services
# This is used to translate addresses to GPS coordinates.
ENABLE_GEOLOCATION=true

# API key for geolocation services
# We use LocationIQ (https://locationiq.com/) to translate addresses to
# latitude/longitude coordinates. We could use Google instead but we don't
# want to give anything to Google, ever.
# LocationIQ offers 10,000 free requests per day.
LOCATION_IQ_API_KEY=MyLocationIQApiKey

# Enable weather on contact profile page
# Weather can only be fetched if we know longitude/latitude - this is why
# you also need to activate the geolocation service above to make it work
ENABLE_WEATHER=true

# Access to weather data from darksky api
# https://darksky.net/dev/register
# Darksky provides an api with 1000 free API calls per day
# You need to enable the weather above if you provide an API key here.
 DARKSKY_API_KEY=MyDarkSkyApiKey
```

**Which version are you using:**
 - A server you maintain yourself (if so, please indicate your current version of Monica)
2.11.2

Can anyone point me out in the right direction to fix this issue?

**Comment by djaiss** [comment]
It's as if the migration that added `temperature_scale` wasn't run.

Can you see if `2018_12_22_021123_add_weather_preferences_to_users` is present in the `migrations` table in your database?

**Comment by brusapa** [comment]
Hello @djaiss , thank you for the answer, as you say it seems like the migration wasn't run, this is the migrations table of my database:

```
mysql> SELECT * FROM migrations;
+-----+--------------------------------------------------------------+-------+
| id  | migration                                                    | batch |
+-----+--------------------------------------------------------------+-------+
|   1 | 2014_10_12_000000_create_users_table                         |     1 |
|   2 | 2014_10_12_100000_create_password_resets_table               |     1 |
|   3 | 2016_06_01_000001_create_oauth_auth_codes_table              |     1 |
|   4 | 2016_06_01_000002_create_oauth_access_tokens_table           |     1 |
|   5 | 2016_06_01_000003_create_oauth_refresh_tokens_table          |     1 |
|   6 | 2016_06_01_000004_create_oauth_clients_table                 |     1 |
|   7 | 2016_06_01_000005_create_oauth_personal_access_clients_table |     1 |
|   8 | 2016_06_07_234741_create_account_table                       |     1 |
|   9 | 2016_06_08_003006_add_account_info_table                     |     1 |
|  10 | 2016_06_08_005413_create_contacts_table                      |     1 |
|  11 | 2016_06_25_224219_create_reminder_type_table                 |     1 |
|  12 | 2016_06_28_191025_create_tasks_table                         |     1 |
|  13 | 2016_06_30_185050_create_notes_table                         |     1 |
|  14 | 2016_07_25_133835_add_width_field                            |     1 |
|  15 | 2016_08_28_122938_create_kids_table                          |     1 |
|  16 | 2016_08_28_215159_create_relations_table                     |     1 |
|  17 | 2016_09_03_202027_add_reminder_id_to_contacts                |     1 |
|  18 | 2016_09_05_134937_add_last_talked_to_field                   |     1 |
|  19 | 2016_09_05_135927_add_people_id_to_contacts                  |     1 |
|  20 | 2016_09_05_145111_add_name_info_to_peoples                   |     1 |
|  21 | 2016_09_06_213550_create_activity_type_table                 |     1 |
|  22 | 2016_09_10_164406_create_jobs_table                          |     1 |
|  23 | 2016_09_10_170122_create_notifications_table                 |     1 |
|  24 | 2016_09_12_014120_create_failed_jobs_table                   |     1 |
|  25 | 2016_09_30_014720_add_kid_to_reminder                        |     1 |
|  26 | 2016_10_15_024156_add_deleted_at_to_users                    |     1 |
|  27 | 2016_10_19_155139_create_cache_table                         |     1 |
|  28 | 2016_10_19_155800_create_sessions_table                      |     1 |
|  29 | 2016_10_21_022941_add_statistics_table                       |     1 |
|  30 | 2016_10_24_013543_add_journal_setting_to_users               |     1 |
|  31 | 2016_10_24_014257_create_journal_tables                      |     1 |
|  32 | 2016_10_28_002518_add_metric_to_settings                     |     1 |
|  33 | 2016_11_01_014353_create_activities_table                    |     1 |
|  34 | 2016_11_01_015957_add_icon_column                            |     1 |
|  35 | 2016_11_03_150307_add_activity_location_to_activities        |     1 |
|  36 | 2016_11_09_013049_add_events_table                           |     1 |
|  37 | 2016_12_08_011555_remove_type_from_notes                     |     1 |
|  38 | 2016_12_13_133945_add_gifts_table                            |     1 |
|  39 | 2016_12_28_150831_change_title_column                        |     1 |
|  40 | 2017_01_14_200815_add_facebook_columns_to_users_table        |     1 |
|  41 | 2017_01_15_045025_add_colors_to_users                        |     1 |
|  42 | 2017_01_22_142645_add_fields_to_contacts                     |     1 |
|  43 | 2017_01_23_043831_change_people_to_contact_for_kids          |     1 |
|  44 | 2017_01_26_013524_change_people_to_significantother          |     1 |
|  45 | 2017_01_26_022852_change_notes_to_contact                    |     1 |
|  46 | 2017_01_26_034553_add_notes_count_to_contact                 |     1 |
|  47 | 2017_01_27_024356_change_people_in_events                    |     1 |
|  48 | 2017_01_28_180156_remove_deleted_at_from_significant_others  |     1 |
|  49 | 2017_01_28_184901_remove_deleted_at_from_kids                |     1 |
|  50 | 2017_01_28_193913_remove_deleted_at_from_notes               |     1 |
|  51 | 2017_01_28_222114_remove_viewed_at_from_contacts             |     1 |
|  52 | 2017_01_29_175146_remove_delete_at_from_activities           |     1 |
|  53 | 2017_01_29_175629_add_number_activities_to_contacts          |     1 |
|  54 | 2017_01_31_025849_add_activity_statistics_table              |     1 |
|  55 | 2017_02_02_232450_add_confirmation                           |     1 |
|  56 | 2017_02_04_225618_change_reminders_table                     |     1 |
|  57 | 2017_02_05_035925_add_gifts_metrics_to_contacts              |     1 |
|  58 | 2017_02_05_041740_change_gifts_table                         |     1 |
|  59 | 2017_02_05_042122_change_people_to_contact_for_gifts         |     1 |
|  60 | 2017_02_07_041607_change_tasks_table                         |     1 |
|  61 | 2017_02_07_051355_add_number_tasks_to_contact                |     1 |
|  62 | 2017_02_08_002251_change_number_tasks_contact                |     1 |
|  63 | 2017_02_08_025358_add_sort_preferences_to_users              |     1 |
|  64 | 2017_02_10_195613_remove_notifications_table                 |     1 |
|  65 | 2017_02_10_214714_remove_people_table                        |     1 |
|  66 | 2017_02_10_215405_remove_entities_table                      |     1 |
|  67 | 2017_02_10_215705_remove_deleted_at_from_contact             |     1 |
|  68 | 2017_02_10_224355_calculate_statistics                       |     1 |
|  69 | 2017_02_11_154900_add_avatars_to_contacts                    |     1 |
|  70 | 2017_02_12_134220_create_entries_table                       |     1 |
|  71 | 2017_05_03_155254_move_significant_other_data                |     1 |
|  72 | 2017_05_04_164723_remove_contact_encryption                  |     1 |
|  73 | 2017_05_04_185921_add_title_to_activities                    |     1 |
|  74 | 2017_05_04_193252_alter_activity_nullable                    |     1 |
|  75 | 2017_05_08_164514_remove_encryption_tasks                    |     1 |
|  76 | 2017_05_30_002239_remove_predefined_reminders                |     1 |
|  77 | 2017_05_30_023116_create_money_table                         |     1 |
|  78 | 2017_06_07_173437_add_multiple_genders_choices               |     1 |
|  79 | 2017_06_10_152945_add_social_networks_to_contacts            |     1 |
|  80 | 2017_06_10_155349_create_currencies_data                     |     1 |
|  81 | 2017_06_11_025227_remove_encryption_journal                  |     1 |
|  82 | 2017_06_11_110735_change_unique_constraint_for_contacts      |     1 |
|  83 | 2017_06_13_035059_remove_gifts_encryption                    |     1 |
|  84 | 2017_06_13_195740_add_company_to_contacts                    |     1 |
|  85 | 2017_06_14_131803_remove_bern_timezone                       |     1 |
|  86 | 2017_06_14_132911_add_zar_currency_to_currencies_table       |     1 |
|  87 | 2017_06_16_215256_add_about_who_to_reminders                 |     1 |
|  88 | 2017_06_17_010900_fix_contacts_table                         |     1 |
|  89 | 2017_06_17_153814_refactor_user_table                        |     1 |
|  90 | 2017_06_19_105842_add_stripe_fields_to_users                 |     1 |
|  91 | 2017_06_20_121345_add_invitations_statistics                 |     1 |
|  92 | 2017_06_22_210813_add_name_order_to_users                    |     1 |
|  93 | 2017_06_27_134704_create_import_table                        |     1 |
|  94 | 2017_06_29_211725_add_import_job_to_statistics               |     1 |
|  95 | 2017_06_29_230523_add_gravatar_url_to_users                  |     1 |
|  96 | 2017_07_02_155736_create_tags_table                          |     1 |
|  97 | 2017_07_04_132743_add_tags_to_statistics                     |     1 |
|  98 | 2017_07_09_164312_update_bad_translation_key                 |     1 |
|  99 | 2017_07_12_014244_create_calls_table                         |     1 |
| 100 | 2017_07_17_005012_drop_reminders_count_from_contacts         |     1 |
| 101 | 2017_07_18_215312_add_danish_kroner_to_currencies_table      |     1 |
| 102 | 2017_07_18_215758_add_indian_rupee_to_currencies_table       |     1 |
| 103 | 2017_07_19_094503_add_brazilian_real_to_currencies           |     1 |
| 104 | 2017_07_22_153209_create_instance_table                      |     1 |
| 105 | 2017_07_26_220021_change_contacts_table                      |     1 |
| 106 | 2017_08_02_152838_change_string_to_boolean_for_reminders     |     1 |
| 107 | 2017_08_06_085629_change_events_data                         |     1 |
| 108 | 2017_08_06_153253_move_kids_to_contacts                      |     1 |
| 109 | 2017_08_16_041431_add_contact_avatar_location                |     1 |
| 110 | 2017_08_21_224835_remove_paid_limitations_for_current_users  |     1 |
| 111 | 2017_09_10_125918_remove_unusued_counters                    |     1 |
| 112 | 2017_09_13_095923_add_tracking_table                         |     1 |
| 113 | 2017_09_13_191714_add_partial_notion                         |     1 |
| 114 | 2017_10_14_083556_change_gift_column_structure               |     1 |
| 115 | 2017_10_17_170803_change_gift_structure                      |     1 |
| 116 | 2017_10_19_134816_create_activity_contact_table              |     1 |
| 117 | 2017_10_19_135215_move_activities_to_pivot_table             |     1 |
| 118 | 2017_10_25_102923_remove_contact_id_activities_table         |     1 |
| 119 | 2017_11_01_122541_add_met_through_to_contacts                |     1 |
| 120 | 2017_11_02_202601_add_is_dead_to_contacts                    |     1 |
| 121 | 2017_11_10_174654_create_contact_fields_table                |     1 |
| 122 | 2017_11_10_181043_migrate_contacts_information               |     1 |
| 123 | 2017_11_10_202620_move_addresses_from_contact_to_addresses   |     1 |
+-----+--------------------------------------------------------------+-------+
123 rows in set (0.00 sec)
```

**Comment by asbiin** [comment]
Did you run `php artisan migrate` or `php artisan monica:update` after updating your instance?

**Comment by brusapa** [comment]
Sorry for the late reply I have been busy and I haven't been able to test this until yesterday.
I just clean the installation by deleting the containers and the images and re-run the docker-compose, now everything works just fine.
I have no idea what I was doing wrong last time.

Thank you for your help!

**Comment by neezer** [comment]
I am experiencing this same issue, also with `docker-compose`. I ran `php artisan setup:production` which ran fine, and I also tried running `php artisan migrate` which errored, saying that the table "photos" already existed. `php artisan monica:update` also ran fine.

The last run migration appears to be `2018_11_15_172333_make_contact_id_nullable_in_tasks`.

I've restarted several times and the problem has not fixed itself, so I'm keen to discover a solution!

Monica (via Docker) v2.11.2
